tests/integration/pkg/helpers/api_rule.go

```diff
@@ -90,17 +91,17 @@ func ApplyApiRuleV2Alpha1(toExecute RetryableApiRule, onRetry RetryableApiRule,
 			if err != nil {
 				return err
 			}
-			if apiStatus.Status.State != "Ready" {
-				log.Println("APIRule status not Ready: " + apiStatus.Status.Description)
-				return errors.New("APIRule status not Ready: " + apiStatus.Status.Description)
+			if apiStatus.Status.State == "Error" {
+				log.Printf("APIRule status is Error: %s", apiStatus.Status.Description)
+				return fmt.Errorf("APIRule status is Error: %s", apiStatus.Status.Description)
 			}
 			return nil
 		}, retryOpts...)
 	}
 	return nil
 }
 
-func ApplyApiRuleV2Alpha1ExpectError(toExecute RetryableApiRule, onRetry RetryableApiRule, k8sClient dynamic.Interface, retryOpts []retry.Option, resources []unstructured.Unstructured, errorMessage string) error {
+func ApplyApiRuleV2Alpha1ExpectStatusAndDesc(toExecute RetryableApiRule, onRetry RetryableApiRule, k8sClient dynamic.Interface, retryOpts []retry.Option, resources []unstructured.Unstructured, status, descMessage string) error {
```

Comment:
This function does exactly the same thing as `ApplyApiRuleV2Alpha1RetryOnError` but the only change is adding the descMessage validation. Please consolidate these functions and adjust the code.